File: leetcode_C++/p2811.cpp
/*
@Author: mrxhar
@Date:   2023-08-29 10:20:50
@Email:  1337904145@qq.com
@Blog:   https://leetcode.cn/problems/check-if-it-is-possible-to-split-array/description/
*/
#include <iostream>
#include <vector>
#include <algorithm>

using std::vector;

class Solution
{
public:
    bool canSplitArray(vector<int> &nums, int m)
    {
        int n = nums.size();
        if (n <= 2)
        {
            return true;
        }

        for (int i = 1; i < n; ++i)
        {
            if (nums[i] + nums[i - 1] >= m)
            {
                return true;
            }
        }
        return false;
    }
};

File: leetcode_py/p2811.py
# 每次拆分都是从两端开始
# 只需要两个相邻的值和大于m
from typing import List


class Solution:
    def canSplitArray(self, nums: List[int], m: int) -> bool:
        if len(nums ) <= 2:
            return True
        for i in range(1, len(nums)):
            if nums[i - 1] + nums[i] >= m:
                return True
        return False